static/js/components/ModifyInstrument.jsx

```diff
@@ -0,0 +1,281 @@
+    return <h3>No instruments available...</h3>;
+  }
+
+  if (enum_types.length === 0) {
+    return (
+      <div>
+        <CircularProgress color="secondary" />
+      </div>
+    );
+  }
+
+  // need to check both of these conditions as selectedAllocationId is
+  // initialized to be null and useEffect is not called on the first
+  // render to update it, so it can be null even if allocationList is not
+  // empty.
+  if (!selectedInstrumentId) {
+    return <h3>No instruments available...</h3>;
+  }
+
+  const api_classnames = [...enum_types.ALLOWED_API_CLASSNAMES];
+  api_classnames.push("");
+  const filters = [...enum_types.ALLOWED_BANDPASSES];
+
+  if (telescopeList.length === 0 || instrumentList.length === 0) {
+    return (
+      <div>
+        <CircularProgress color="secondary" />
+      </div>
+    );
+  }
+
+  const telLookUp = {};
+  // eslint-disable-next-line no-unused-expressions
+  telescopeList?.forEach((tel) => {
+    telLookUp[tel.id] = tel;
+  });
+
+  const instLookUp = {};
+  // eslint-disable-next-line no-unused-expressions
+  instrumentList?.forEach((instrumentObj) => {
+    instLookUp[instrumentObj.id] = instrumentObj;
+  });
+
+  const handleSelectedInstrumentChange = (e) => {
+    setSelectedInstrumentId(e.target.value);
+  };
+
+  function validate(formData, errors) {
+    const oldFilters = [];
+    instLookUp[selectedInstrumentId].filters?.forEach((filter) => {
+      oldFilters.push(filter);
+    });
+    const oldFiltersUnique = [...new Set(oldFilters)];
+
+    const newFilters = [];
+    formData.filters?.forEach((filter) => {
+      newFilters.push(filter);
+    });
+    const newFiltersUnique = [...new Set(newFilters)];
+
+    const result = oldFiltersUnique.every((val) =>
+      newFiltersUnique.includes(val)
+    );
+    if (errors && formData.filters && !result) {
+      errors.filters.addError("New filter list must contain old filter list.");
+    }
+    if (errors && formData.api_classname && formData.api_classname.length > 1) {
+      errors.api_classname.addError("Must only choose one API class.");
+    }
+    if (
+      errors &&
+      formData.api_classname_obsplan &&
+      formData.api_classname_obsplan.length > 1
+    ) {
+      errors.api_classname_obsplan.addError("Must only choose one API class.");
+    }
+    return errors;
+  }
+
+  const instrumentFormSchema = {
+    type: "object",
+    properties: {
+      filters: {
+        type: "array",
+        items: {
+          type: "string",
+          enum: filters,
+        },
+        uniqueItems: true,
+        title: "Filter list",
+      },
+      api_classname: {
+        type: "array",
+        items: {
+          type: "string",
+          enum: api_classnames,
+        },
+        uniqueItems: true,
+        title: "API Classname",
+      },
+      api_classname_obsplan: {
+        type: "array",
+        items: {
+          type: "string",
+          enum: api_classnames,
```

Comment:
Is this supposed to be the same enum types for `api_classname` and `api_classname_obsplan`?

Reply:
yeah they are one list, all the facility apis.